### PR TITLE
Closes #9236: Lazy FxaAccountManager initialization

### LINKED ISSUE #9236: BackgroundServices should not be initialized on startup
Currently we're initializing BackgroundServices (and kicking off FxaAccountManager's init) on startup. Strictly speaking, we don't need to do this so early on. Let's understand all of the touch points, and how we can defer this work.

### PR DESCRIPTION
**Pull request by grigoryk**
<del>This is a rolling WIP as I'm working through https://github.com/mozilla-mobile/fenix/issues/9236</del>

This should be ready for review; commit series is the best way to follow along with the changes.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by grigoryk** [comment]
All failures in `ui-test-x86-debug` have `RunWhenReadyQueue` in their path, so let's fix them first.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -384,19 +378,25 @@ class HomeFragment : Fragment() {
...
-            override fun onAuthenticated(account: OAuthAccount, authType: AuthType) {
-                if (authType != AuthType.Existing) {
-                    view?.let {
-                        FenixSnackbar.make(it, Snackbar.LENGTH_SHORT)
-                            .setText(it.context.getString(R.string.onboarding_firefox_account_sync_is_on))
-                            .setAnchorView(toolbarLayout)
-                            .show()
+        requireComponents.backgroundServices.accountManagerAvailableQueue.runIfReadyOrQueue {
+            // By the time this code runs, we may not be attached to a context.
+            if ((this@HomeFragment).context == null) {
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: We could use https://github.com/mozilla-mobile/fenix/blob/be7bd527f68a4a128be5f3f27f93650da0fc6f3e/app/src/main/java/org/mozilla/fenix/addons/Extensions.kt#L39

Comment (csadilek) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: or we could combine the two.

File: app/src/main/java/org/mozilla/fenix/utils/RunWhenReadyQueue.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import kotlinx.coroutines.launch
+import java.util.concurrent.CopyOnWriteArrayList
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * A queue that acts as a gate, either executing tasks right away if the queue is marked as "ready",
+ * i.e. gate is open, or queues them to be executed whenever the queue is marked as ready in the
+ * future, i.e. gate becomes open.
+ */
+class RunWhenReadyQueue {
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/utils/RunWhenReadyQueue.kt]: Let's write a unit test or file a follow up :).

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/utils/RunWhenReadyQueue.kt]: Filed https://github.com/mozilla-mobile/android-components/issues/6460

**Pull request merged**